hibernate-core/src/main/java/org/hibernate/internal/StatelessSessionImpl.java

```diff
@@ -558,20 +556,18 @@ public List list(String query, QueryParameters queryParameters) throws Hibernate
 		checkOpen();
 		queryParameters.validateParameters();
 		HQLQueryPlan plan = getQueryPlan( query, false );
-		boolean success = false;
-		List results = Collections.EMPTY_LIST;
+		List results;
 		try {
 			results = plan.performList( queryParameters, this );
-			success = true;
 		}
 		finally {
-			afterOperation( success );
+			afterOperation();
 		}
 		temporaryPersistenceContext.clear();
 		return results;
 	}
 
-	public void afterOperation(boolean success) {
+	public void afterOperation() {
```

Comment:
Maintaining public interface backward compatibility is one of the few lessons I learned here. As a general guideline, we should not touch public method (even protected method) and if you need to introduce new method (or generalize existing one by introducing new parameter), you are supposed to rely on JDK's default interface method to maintain backward compatibility.

Reply:
It is not enough to make the PR building green. Our client might be using the public method and non-compatible code change might break their compilation. That being said, dropping the unnecessary parameter might be a good point in v6, if I were not mistaken.

Reply:
I agree with what you said, and that is exactly what I was concerned about in this change. However, since that is something that we cannot change, we can move forward with closing this PR as changing the public method might have an impact on the clients extending the StatelessSessionImpl

Thoughts?

Reply:
One thing that had got me to refactor this code were the multiple occurrences of a boolean variable named success in the same class. It was being passed as a parameter in this method `afterOperation()`

How do you think we can improve that, if we don't refactor this method though? 
And if not, should we let it stay as it is as well?

Reply:
Before closing the PR, maybe it is worthwhile to investigate whether it is a good idea to merge into wip/v6.0? @sebersole is the go-to-person.

Reply:
Sure. Merging it into wip/v6.0 sounds like a good idea to me. If @sebersole thinks that is doable and makes sense, I can then update the base branch for this PR.